amaranth/hdl/_xfrm.py

```diff
@@ -277,6 +277,23 @@ def on_fragment(self, fragment):
             new_fragment = Instance(fragment.type, src_loc=fragment.src_loc)
             new_fragment.parameters = OrderedDict(fragment.parameters)
             self.map_named_ports(fragment, new_fragment)
+        elif isinstance(fragment, IOBufferInstance):
+            if hasattr(self, "on_value"):
```

Comment:
This is criminal, but really it's just more motivation to get rid of transformers.

Reply:
I mean, we do the exact same thing for `Instance` and `MemoryInstance`...

Reply:
It's criminal there too!